From b0a28ece87e2c10dc1d67023a6c075992e8f3da4 Mon Sep 17 00:00:00 2001
From: Thomas Willingham <beardyunixer@beardyunixer.com>
Date: Sat, 14 Feb 2015 16:13:17 +0000
Subject: [PATCH] More dead stuff

---
 doc/external-resource-links.bb | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/doc/external-resource-links.bb b/doc/external-resource-links.bb
index ca1d6f5..2e6df63 100644
--- a/doc/external-resource-links.bb
+++ b/doc/external-resource-links.bb
@@ -8,16 +8,11 @@
 [*][url=https://github.com/tonybaldwin/redmatrixthemes/]nubasic[/url]
 
 [b]Third-Party Addons[/b]
-[*][url=http://gitweb.whogotzot.com/red-addons-extra/tree]Additional RedMatrix addons[/url]
 [*][url=https://abcentric.net/git/abcjsplugin.git]ABCjs integration - display scores in posts (WIP)[/url]
-[*][url=http://gitweb.whogotzot.com/extra_dir_fns/tree]Extra directory functions[/url]
 [b]Related Projects[/b]
 
 [*][url=https://addons.mozilla.org/en-US/firefox/addon/redshare/]Redshare for Firefox[/url]
 [*][url=https://github.com/cvogeley/red-for-android]Red for Android[/url]
 [*][url=https://github.com/zzottel/feed2red]feed2red.pl (posts Atom/RSS feeds to channel)[/url]
-[*][url=http://gitweb.whogotzot.com/red-stable/tree]Red Stable - up-to-date-but-not-bleeding-edge RedMatrix distribution[/url]
-[b]Utilities[/b]
-[*][url=http://gitweb.whogotzot.com/debian-install-script]Debian Install Script[/url]
 
 #include doc/macros/main_footer.bb;
-- 
1.9.1

